FreeType/Error/Values.hsc

```diff
@@ -0,0 +1,299 @@
+      , FT_Err_Array_Too_Large
+      , FT_Err_Missing_Module
+      , FT_Err_Missing_Property
+      , FT_Err_Invalid_Glyph_Index
+      , FT_Err_Invalid_Character_Code
+      , FT_Err_Invalid_Glyph_Format
+      , FT_Err_Cannot_Render_Glyph
+      , FT_Err_Invalid_Outline
+      , FT_Err_Invalid_Composite
+      , FT_Err_Too_Many_Hints
+      , FT_Err_Invalid_Pixel_Size
+      , FT_Err_Invalid_Handle
+      , FT_Err_Invalid_Library_Handle
+      , FT_Err_Invalid_Driver_Handle
+      , FT_Err_Invalid_Face_Handle
+      , FT_Err_Invalid_Size_Handle
+      , FT_Err_Invalid_Slot_Handle
+      , FT_Err_Invalid_CharMap_Handle
+      , FT_Err_Invalid_Cache_Handle
+      , FT_Err_Invalid_Stream_Handle
+      , FT_Err_Too_Many_Drivers
+      , FT_Err_Too_Many_Extensions
+      , FT_Err_Out_Of_Memory
+      , FT_Err_Unlisted_Object
+      , FT_Err_Cannot_Open_Stream
+      , FT_Err_Invalid_Stream_Seek
+      , FT_Err_Invalid_Stream_Skip
+      , FT_Err_Invalid_Stream_Read
+      , FT_Err_Invalid_Stream_Operation
+      , FT_Err_Invalid_Frame_Operation
+      , FT_Err_Nested_Frame_Access
+      , FT_Err_Invalid_Frame_Read
+      , FT_Err_Raster_Uninitialized
+      , FT_Err_Raster_Corrupted
+      , FT_Err_Raster_Overflow
+      , FT_Err_Raster_Negative_Height
+      , FT_Err_Too_Many_Caches
+      , FT_Err_Invalid_Opcode
+      , FT_Err_Too_Few_Arguments
+      , FT_Err_Stack_Overflow
+      , FT_Err_Code_Overflow
+      , FT_Err_Bad_Argument
+      , FT_Err_Divide_By_Zero
+      , FT_Err_Invalid_Reference
+      , FT_Err_Debug_OpCode
+      , FT_Err_ENDF_In_Exec_Stream
+      , FT_Err_Nested_DEFS
+      , FT_Err_Invalid_CodeRange
+      , FT_Err_Execution_Too_Long
+      , FT_Err_Too_Many_Function_Defs
+      , FT_Err_Too_Many_Instruction_Defs
+      , FT_Err_Table_Missing
+      , FT_Err_Horiz_Header_Missing
+      , FT_Err_Locations_Missing
+      , FT_Err_Name_Table_Missing
+      , FT_Err_CMap_Table_Missing
+      , FT_Err_Hmtx_Table_Missing
+      , FT_Err_Post_Table_Missing
+      , FT_Err_Invalid_Horiz_Metrics
+      , FT_Err_Invalid_CharMap_Format
+      , FT_Err_Invalid_PPem
+      , FT_Err_Invalid_Vert_Metrics
+      , FT_Err_Could_Not_Find_Context
+      , FT_Err_Invalid_Post_Table_Format
+      , FT_Err_Invalid_Post_Table
+      , FT_Err_DEF_In_Glyf_Bytecode
+      , FT_Err_Missing_Bitmap
+      , FT_Err_Syntax_Error
+      , FT_Err_Stack_Underflow
+      , FT_Err_Ignore
+      , FT_Err_No_Unicode_Glyph_Name
+      , FT_Err_Glyph_Too_Big
+      , FT_Err_Missing_Startfont_Field
+      , FT_Err_Missing_Font_Field
+      , FT_Err_Missing_Size_Field
+      , FT_Err_Missing_Fontboundingbox_Field
+      , FT_Err_Missing_Chars_Field
+      , FT_Err_Missing_Startchar_Field
+      , FT_Err_Missing_Encoding_Field
+      , FT_Err_Missing_Bbx_Field
+      , FT_Err_Bbx_Too_Big
+      , FT_Err_Corrupted_Font_Header
+      , FT_Err_Corrupted_Font_Glyphs
+     :: FT_Error
+pattern FT_Err_Ok                            = #const FT_Err_Ok
```

Comment:
From the get go I decided to use patterns simply because all of those names have `FT` capitalized in the FreeType library and `gl` already does export every value of this kind as patterns.

However the part where you make everything `(Eq a, Num a) => a` is a bit wacky here, because certain definitions (e.g. [FT_Glyph_Format](https://www.freetype.org/freetype2/docs/reference/ft2-basic_types.html#ft_glyph_format)) actually get used as types later on (e.g. in [FT_GlyphSlotRec](https://www.freetype.org/freetype2/docs/reference/ft2-base_interface.html#ft_glyphslotrec), while some do not.

So I guess I should leave the types be and set every pattern to `(Eq a, Num a) => a` nonetheless.

Reply:
OK -- I would use `(Eq a, Num a) => a` where it makes sense for consistency with other packages and keep it as is otherwise. For things like `FT_Error` I would expect a similar type enum like `FT_Glyph_Format`. Looking over the files again, you seem to have used good judgement where applicable, so I trust you'll make the appropriate changes. :)